ktor-server/ktor-server-tests/jvm/test/io/ktor/server/sessions/SessionTestJvm.kt

```diff
@@ -1,198 +1,192 @@
 
-@Suppress("ReplaceSingleLineLet", "DEPRECATION")
+@Suppress("ComplexRedundantLet")
 class SessionTestJvm {
     private val cookieName = "_S" + Random.nextInt(100)
 
     @Test
-    fun testSessionByValueMac() {
+    fun testSessionByValueMac() = testApplication {
         val key = hex("03515606058610610561058")
-        withTestApplication {
-            application.install(Sessions) {
-                cookie<TestUserSession>(cookieName) {
-                    transform(SessionTransportTransformerMessageAuthentication(key))
-                }
-            }
 
-            commonSignedChecks()
+        install(Sessions) {
+            cookie<TestUserSession>(cookieName) {
+                transform(SessionTransportTransformerMessageAuthentication(key))
+            }
         }
+
+        commonSignedChecks()
     }
 
     @Test
-    fun testSessionEncrypted() {
+    fun testSessionEncrypted() = testApplication {
         val encryptKey = hex("00112233445566778899aabbccddeeff")
         val signKey = hex("02030405060708090a0b0c")
         val forcedIvForTesting = hex("00112233445566778899aabbccddeeff")
 
-        withTestApplication {
-            application.install(Sessions) {
-                cookie<TestUserSession>(cookieName) {
-                    transform(SessionTransportTransformerEncrypt(encryptKey, signKey, { forcedIvForTesting }))
-                }
+        install(Sessions) {
+            cookie<TestUserSession>(cookieName) {
+                transform(SessionTransportTransformerEncrypt(encryptKey, signKey, { forcedIvForTesting }))
             }
+        }
 
-            application.routing {
-                get("/3") {
-                    call.sessions.set(TestUserSession("id2", emptyList()))
-                    call.respondText("ok")
-                }
-                get("/4") {
-                    call.respondText("ok:" + call.sessions.get<TestUserSession>()?.userId)
-                }
+        routing {
+            get("/3") {
+                call.sessions.set(TestUserSession("id2", emptyList()))
+                call.respondText("ok")
             }
-
-            handleRequest(HttpMethod.Get, "/3").let { call ->
-                val sessionCookie = call.response.cookies[cookieName]
-                assertEquals(
-                    "00112233445566778899aabbccddeeff/" +
-                        "9b8cf9900ff0f63849cb7a2ad71ed8a1ac9f0e2735492d2a5129cf278d70ab27:" +
-                        "40af9f393a00034b53d2267251047259005bdf5ff104eede599149e80e6fee13",
-                    sessionCookie!!.value
-                )
+            get("/4") {
+                call.respondText("ok:" + call.sessions.get<TestUserSession>()?.userId)
             }
+        }
 
-            handleRequest(HttpMethod.Get, "/4") {
-                addHeader(HttpHeaders.Cookie, "$cookieName=INVALID")
-            }.let { call ->
-                assertEquals("ok:null", call.response.content)
-            }
+        commonSignedChecks()
```

Comment:
I don't like that it is important to place `commonSignedChecks` before first client call (before application is started). Do you have any ideas on how to make the tests less fragile and keep them clear?

Reply:
Hmm I don't think there's an easy way to resolve this.  I'd say leave it for now.